be/src/pipeline/pipeline_task.cpp

```diff
@@ -291,24 +291,19 @@ void PipelineTask::terminate() {
     auto fragment = _fragment_context.lock();
     if (!is_finalized() && fragment) {
         DCHECK(_wake_up_early || fragment->is_canceled());
-        for (auto* dep : _spill_dependencies) {
-            dep->set_always_ready();
-        }
-
-        for (auto* dep : _filter_dependencies) {
-            dep->set_always_ready();
-        }
-        for (auto& deps : _read_dependencies) {
-            for (auto* dep : deps) {
-                dep->set_always_ready();
-            }
-        }
-        for (auto* dep : _write_dependencies) {
-            dep->set_always_ready();
-        }
-        for (auto* dep : _finish_dependencies) {
-            dep->set_always_ready();
-        }
+        std::for_each(_spill_dependencies.begin(), _spill_dependencies.end(),
+                      [&](Dependency* dep) { dep->set_always_ready(); });
```

Comment:
maybe we can reuse the lambda [&](Dependency* dep) { dep->set_always_ready(); }?